arroyo-console/src/routes/pipelines/PipelineConfigModal.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from "react";
+import {
+  Alert,
+  AlertDescription,
+  AlertIcon,
+  Button,
+  FormControl,
+  FormHelperText,
+  FormLabel,
+  Input,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+} from "@chakra-ui/react";
+
+export interface PipelineConfigModalProps {
+  isOpen: boolean;
+  parallelism: number;
+  onClose: () => void;
+  updateJobParallelism: (parallelism: number) => void;
+}
+
+const PipelineConfigModal: React.FC<PipelineConfigModalProps> = ({
+  isOpen,
+  parallelism,
+  onClose,
+  updateJobParallelism,
+}) => {
+  const [parallelismInputValue, setParallelismInputValue] = useState<number | undefined>(
+    parallelism
+  );
+
+  const onConfirm = () => {
+    if (parallelismInputValue) {
+      updateJobParallelism(parallelismInputValue);
+    }
+    onClose();
+  };
+
+  const pipelineRequiresRestart = parallelismInputValue != parallelism;
+
+  let changeAlert = <></>;
+  if (pipelineRequiresRestart) {
+    changeAlert = (
+      <Alert status="warning">
+        <AlertIcon />
+        <AlertDescription>These changes require a pipeline restart.</AlertDescription>
+      </Alert>
+    );
+  }
+
+  const parallelismInput = (
+    <Input
```

Comment:
Should this be a [number input](https://chakra-ui.com/docs/components/number-input)?

Reply:
Yes, done. I had to manually set the background color, though. Maybe there's a bug with the dark theme.